portfolio/src/main/java/com/google/sps/servlets/UserData.java

```diff
@@ -0,0 +1,14 @@
+package com.google.sps.servlets;
+
+class UserData {
+    private boolean isLoggedIn = false;
+    private String email;
+
+    UserData(String email) {
+        if (!email.isEmpty()) {
```

Comment:
While this works with your current logic, I think it's actually too constraining. It'd be best if you'd just pass in `isLoggedIn` as a separate variable. After all, a user that's NOT logged in could still have user data (consider the case of an admin looking up some users on some admin UI).

Reply:
Done.